$NetBSD: patch-aa,v 1.7 2016/01/19 03:44:03 dbj Exp $

Support for dragonfly
Shell portability fixes
Use X11 and OpenGL Libs on Darwin

--- configure.orig	2011-03-22 12:16:57.000000000 +0000
+++ configure
@@ -1962,7 +1962,7 @@ _ACEOF
 
     DEFAULT_DEFAULT_wxUSE_MOTIF=1
   ;;
-  *-*-freebsd*)
+  *-*-freebsd* | *-*-dragonfly*)
     USE_BSD=1
     USE_FREEBSD=1
     cat >>confdefs.h <<\_ACEOF
@@ -16139,7 +16139,7 @@ retest_macosx_linking=no
 
 #if test "$wxUSE_MAC" = 1; then
     if test "x$wxUSE_UNIVERSAL_BINARY" != xno ; then
-                        if test "x$wxUSE_UNIVERSAL_BINARY" == xyes; then
+                        if test "x$wxUSE_UNIVERSAL_BINARY" = xyes; then
             # Implicitly turn on the new --with-macosx-sdk using the default
             # SDK which provides the behavior this option has always had.
             if test "x$wxUSE_MACOSX_SDK" = "x"; then
@@ -16221,7 +16221,7 @@ echo $ECHO_N "checking SDK deployment ve
         MACOSX_SDK_PLIST_VERSION_MIN=`defaults read "$wxUSE_MACOSX_SDK/SDKSettings" buildSettings | grep '^ *"\{0,1\}MACOSX_DEPLOYMENT_TARGET"\{0,1\} *= *"\{0,1\}[^"]*"\{0,1\}; *$' | sed 's/^ *"\{0,1\}MACOSX_DEPLOYMENT_TARGET"\{0,1\} *= *"\{0,1\}\([^"]*\)"\{0,1\} *; *$/\1/'`
 
         # If that failed, try again with the new key
-        if test "x$MACOSX_SDK_PLIST_VERSION_MIN" == "x"; then
+        if test "x$MACOSX_SDK_PLIST_VERSION_MIN" = "x"; then
 
             MACOSX_SDK_PLIST_VERSION_MIN=`defaults read "$wxUSE_MACOSX_SDK/SDKSettings" DefaultProperties | grep '^ *"\{0,1\}MACOSX_DEPLOYMENT_TARGET"\{0,1\} *= *"\{0,1\}[^"]*"\{0,1\}; *$' | sed 's/^ *"\{0,1\}MACOSX_DEPLOYMENT_TARGET"\{0,1\} *= *"\{0,1\}\([^"]*\)"\{0,1\} *; *$/\1/'`
 
@@ -24948,6 +24948,7 @@ SEARCH_INCLUDE="\
                               \
     /usr/include/Xm           \
                               \
+    /usr/X11R7/include        \
     /usr/X11R6/include        \
     /usr/X11R6.4/include      \
                               \
@@ -25055,11 +25056,12 @@ fi
         wx_cv_std_libpath="lib";
         ;;
 esac
+wx_cv_std_libpath="lib"
 
 echo "$as_me:$LINENO: result: $wx_cv_std_libpath" >&5
 echo "${ECHO_T}$wx_cv_std_libpath" >&6
 
-SEARCH_LIB="`echo "$SEARCH_INCLUDE" | sed s@include@$wx_cv_std_libpath@g` /usr/$wx_cv_std_libpath"
+SEARCH_LIB="`echo "$SEARCH_INCLUDE" | sed s@include@$wx_cv_std_libpath@g` /usr/$wx_cv_std_libpath @PREFIX@/lib"
 
 if test "$build" != "$host" -a "$GCC" = yes; then
                     if cross_root=`$CC -print-prog-name=ld 2>/dev/null`; then
@@ -33191,7 +33193,7 @@ echo "$as_me: WARNING: wxGLCanvas not im
     elif test "$wxUSE_MSW" = 1; then
         OPENGL_LIBS="-lopengl32 -lglu32"
     elif test "$USE_DARWIN" = 1; then
-        OPENGL_LIBS=""
+        OPENGL_LIBS="-L/usr/X11/lib -lGL -lGLU"
     else
                         echo "$as_me:$LINENO: checking for OpenGL headers" >&5
 echo $ECHO_N "checking for OpenGL headers... $ECHO_C" >&6
@@ -39590,6 +39592,9 @@ echo "$as_me: WARNING: BeOS threads are 
                     THREAD_OPTS="-mt $THREAD_OPTS"
                 fi
                 ;;
+	  *-*-dragonfly*)
+                THREAD_OPTS="-pthread $THREAD_OPTS"
+		;;
           *-*-freebsd*)
                                                 THREAD_OPTS="-kthread lthread $THREAD_OPTS c_r"
                 ;;
@@ -39723,7 +39728,7 @@ echo $ECHO_N "checking if more special f
                                                                                                                 LDFLAGS="-L/usr/lib/threads $LDFLAGS"
                 flag="-D_THREAD_SAFE"
                 ;;
-            *-freebsd*)
+            *-freebsd*|*-*-dragonfly*)
                 flag="-D_THREAD_SAFE"
                 ;;
             *-hp-hpux* )
@@ -48648,7 +48653,7 @@ echo "${ECHO_T}$bakefile_cv_gcc31" >&6
             SHARED_LD_CXX="\${CXX} -dynamiclib -single_module -headerpad_max_install_names -o"
         fi
 
-        if test "x$GCC" == "xyes"; then
+        if test "x$GCC" = "xyes"; then
             PIC_FLAG="-dynamic -fPIC"
         fi
         if test "x$XLCC" = "xyes"; then
@@ -49215,6 +49220,7 @@ EOF
 
       powerpc-apple-macos* | \
       *-*-freebsd* | *-*-openbsd* | *-*-netbsd* | *-*-k*bsd*-gnu | \
+      *-*-dragonfly* | \
       *-*-mirbsd* | \
       *-*-sunos4* | \
       *-*-osf* | \
@@ -49258,7 +49264,7 @@ echo "$as_me: error: unknown system type
 
     case "${BAKEFILE_HOST}" in
       *-*-linux* | *-*-freebsd* | *-*-openbsd* | *-*-netbsd* | \
-      *-*-k*bsd*-gnu | *-*-mirbsd* )
+      *-*-k*bsd*-gnu | *-*-mirbsd* | *-*-dragonfly*)
         if test "x$SUNCXX" = "xyes"; then
             SONAME_FLAG="-h "
         else
